packages/redux/src/orders/selectors.ts

```diff
@@ -292,88 +333,94 @@ export const getOrder: (
     (_, orderId) => orderId,
     (state: StoreState) => getEntities(state, 'orders'),
     getOrderItems,
-    getMerchants,
-    getCategories,
-    getBrands,
   ],
-  (orderId, orders, orderItems, merchants, categories, brands) => {
-    return denormalizeOrder(
-      orderId,
-      orders,
-      orderItems,
-      merchants,
-      categories,
-      brands,
-    );
+  (orderId, orders, orderItems) => {
+    return denormalizeOrder(orderId, orders, orderItems);
   },
 );
 
 /**
- * Returns all return options from a specific order and optionally a merchant.
+ * Returns all return options from a specific order.
  *
  * @param state      - Application state.
  * @param orderId    - Order id.
  * @param merchantId - Merchant id.
  *
  * @returns List of return options objects.
  */
-// export const getOrderReturnOptions: (
-//   state: StoreState,
-//   orderId: Order['id'],
-//   merchantId?: MerchantEntity['id'],
-// ) => ReturnOptionEntityDenormalized[] | null | undefined = createSelector(
-//   [
-//     (state: StoreState, orderId: Order['id']) => getOrder(state, orderId),
-//     (_, orderId: Order['id'], merchantId?: MerchantEntity['id']) => ({
-//       orderId,
-//       merchantId,
-//     }),
-//   ],
-//   (order, { merchantId }) => {
-//     if (!order) {
-//       return undefined;
-//     }
-
-//     if (
-//       !order.returnOptions ||
-//       order.returnOptions.length === 0 ||
-//       !merchantId
-//     ) {
-//       return order.returnOptions;
-//     }
-
-//     return order.byMerchant[merchantId]?.returnOptions;
-//   },
-// );
-
-/**
- * Returns all the merchants from a specific order.
+export const getOrderReturnOptions: (
+  state: StoreState,
+  orderId: Order['id'],
+) =>
+  | Record<
+      MerchantOrderReturnOptions['merchantOrderId'],
+      MerchantOrderReturnOptions
+    >
+  | undefined = createSelector(
+  [
+    (state: StoreState, orderId: Order['id']) => getOrder(state, orderId),
+    (state: StoreState) => getEntities(state, 'returnOptions'),
+    getMerchants,
+  ],
+  (order, returnOptions, merchants) => {
+    if (!order) {
+      return undefined;
+    }
+
+    if (!returnOptions) {
+      return undefined;
+    }
+
+    const result: Record<
+      MerchantOrderReturnOptions['merchantOrderId'],
+      MerchantOrderReturnOptions
+    > = {};
```

Comment:
There is a problem with this return value. When there are no return options entities, this selector will return undefined, which is good, because the caller can then infer that no request was made to fetch the return options and then he can make the request. The problem is when you invoke this selector with another orderId without removing the previous order return options entities. It will return an empty object, but no request was made to the server to fetch the order return options, so we will end up returning an invalid response. We need to change this to sync with the result of the fetch order return options action, which will need changes to the reducer to store the result as well to have it work similarly to the way the `getUserOrdersResult` works.

Reply:
Done, please validate the solution 👍